### PR TITLE
Diminish Database pollution with testing values

### PR DESCRIPTION
**Pull request by Warhedgehog**

### CODE REVIEW

**Review by RafaelRamirezS — APPROVED** [review]
Good code cleanup, nothing to add

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]
You should try to fix the CC issues

File: app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java [hunk 1]
```diff
@@ -237,6 +237,20 @@ public void addGameListener(@Nullable Game game, @Nullable ValueEventListener l)
         }
     }
 
+    public void removeGameListener(Game game, ValueEventListener listener){
+        if (game == null) {
+            throw new IllegalArgumentException("game should not be null.");
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java:242]: These checks should be tested

File: app/src/main/java/sdp/moneyrun/database/UserDatabaseProxy.java [hunk 2]
```diff
@@ -47,6 +48,16 @@ public void putUser(@Nullable User user) {
         usersRef.child(String.valueOf(user.getUserId())).setValue(user);
     }
 
+    public void putUser(User user, OnCompleteListener listener){
+        if(user == null){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/database/UserDatabaseProxy.java:52]: Same here

File: app/src/main/java/sdp/moneyrun/game/Game.java [hunk 3]
```diff
@@ -412,6 +413,26 @@ public void setPlayers(@Nullable List<Player> players, boolean forceLocal) {
 
     }
 
+    public void setPlayers(List<Player> players, boolean forceLocal, OnCompleteListener listener) {
+        if (players == null) {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/game/Game.java:417]: Same here as well

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]

File: app/src/main/java/sdp/moneyrun/game/Game.java (hunk 3 above)
Comment (Warhedgehog) [reply app/src/main/java/sdp/moneyrun/game/Game.java:417]: Same here

File: app/src/main/java/sdp/moneyrun/database/UserDatabaseProxy.java (hunk 2 above)
Comment (Warhedgehog) [reply app/src/main/java/sdp/moneyrun/database/UserDatabaseProxy.java:52]: I didn't write this, I'm pretty sure

File: app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java (hunk 1 above)
Comment (Warhedgehog) [reply app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java:242]: Agreed

**Comment by Warhedgehog** [comment]
> You should try to fix the CC issues

The first two is about code I did not write. The last two should be discarded IMO

**Review by Alvaro2112 — APPROVED** [review]

**Pull request merged**